numba/cuda/compiler.py

```diff
@@ -479,8 +481,11 @@ def __init__(self, py_func, argtypes, link=None, debug=False, inline=False,
         }
 
         tgt_ctx = cres.target_context
+        filename = cres.type_annotation.filename
```

Comment:
It's weird that I have to get the source file name from the type annotation at this stage, but I think the function IR is now lost (it was never saved after the pipeline finished).

Reply:
It's probably in the `self.py_func.__code__.co_filename` and `cres.fndesc.lookup_module.__file__` too if they are preferable? IIRC the compile result doesn't hold reference to the IR as the IR could hold reference to e.g. globals, some of which might be large (e.g. NumPy arrays) and this would create memory pressure etc.